# Coles Online Interstate Differences - Exploratory Analysis

## Purpose

The objective of this exploratory analysis is to ascertain whether Coles charges different prices for the same product in different states. This would imply that Coles passes on supply chain costs to customers who are in areas further away from their main distribution centres. It may also give us a very general idea of where the main distribution centres are located. Insights gained from this analysis may provide a more granular picture of Coles' pricing strategy - which may be useful in formulating pricing and/or market segmentation strategies to compete with Coles.

If price differences are not found between major cities and remote regions of Australia, that would imply that customers in major city centres are _de facto_ subsidising remote area customers' online purchases from Coles.

Date data extracted: _26 January 2021_ 

Coles Online Geolocations: 
* Adelaide SA 
* Melbourne VIC

### Loading data into Pandas and preliminary data scrubbing 


Raw data scraped from Coles often has many data integrity issues. The most common ones being:
* Unavailable Products are listed
* Products listed as available have no price tags (i.e. it's actually unavailable but their system is either bugged or someone didn't make the correct changes on their internal system)
* Duplicate products (i.e. same SKU from links with promotional suffix in the URL which escaped my scraper's filtering)

Below, basic data scrubbing was applied while reading in the CSV files into Pandas.

* Replaced 'NaN' values with an empty string.
* Removed duplicate items with identical SKUs
* Date column removed - only useful in timeseries analysis
* Applied filter to only load in products with the 'Available' tag and a 'Current Price' value

In [49]:
import pandas as pd

df_adl = pd.read_csv('sauce/Jan 26 Adelaide SA.csv').fillna('').drop_duplicates(subset=['SKU']).drop(columns=['Date'])
df_adl = df_adl.loc[(df_adl['Availability'] == 'Available') & (df_adl['Current Price'] != '')]

df_mel = pd.read_csv('sauce/Jan 26 Melbourne VIC.csv').fillna('').drop_duplicates(subset=['SKU']).drop(columns=['Date'])
df_mel = df_mel.loc[(df_mel['Availability'] == 'Available') & (df_mel['Current Price'] != '')]

Let's double check that Coles indeed uses the same SKU numbers for the same products across Australia.

To do this, we'll sort the SKU column and eyeball the first few items of both dataframes.

In [50]:
# Adelaide
df_adl.sort_values(by='SKU').head()

SKU            Brand                    Product Name  \
8897   1000301P       Real Foods         Tasty Cheese Corn Thins   
20151    10016P  Victoria Bitter                       Can 375mL   
19012   100537P        Steradent        Powder Denture Cleansing   
7076   1005430P     Coles Finest     Angus Beef Eye Fillet Steak   
7404   1005611P     Coles Finest  Angus Beef Scotch Fillet Steak   

      Current Price  Price per unit          Unit Size Previous Price  \
8897           1.80  $1.44 per 100G               125g            2.0   
20151         58.00    $6.44 per 1L            24 Pack                  
19012          5.95  $2.98 per 100G               200g                  
7076          16.25  $65.00 per 1Kg  approx. 250g pack                  
7404          17.55  $50.00 per 1Kg  approx. 351g each                  

      On Special  \
8897    $pecials   
20151              
19012              
7076               
7404               

                                                              Suppliers  \
8897                                              ['731170 REAL FOODS']   
20151                                                  ['9393 CUB P/L']   
19012                           ['672882 RECKITT BENCKISER (AUST) P/L']   
7076   ['963906 COLES MT PROD - C9687MT - MAP WA', '964437 MAP WA P/L']   
7404   ['964437 MAP WA P/L', '963906 COLES MT PROD - C9687MT - MAP WA']   

                  Category Availability  \
8897   Summer Entertaining    Available   
20151               Liquor    Available   
19012      Health & Beauty    Available   
7076        Meat & Seafood    Available   
7404        Meat & Seafood    Available   

                                                                                                    URL  
8897            https://shop.coles.com.au/a/churchill-centre/product/real-foods-corn-thins-tasty-cheese  
20151            https://shop.coles.com.au/a/churchill-centre/product/victoria-bitter-375ml-cans-10016p  
19012           https://shop.coles.com.au/a/churchill-centre/product/steradent-denture-cleansing-powder  
7076      https://shop.coles.com.au/a/churchill-centre/product/coles-finest-steak-angus-beef-eye-fillet  
7404   https://shop.coles.com.au/a/churchill-centre/product/coles-finest-steak-angus-beef-scotch-fillet

In [51]:
# Melbourne
df_mel.sort_values(by='SKU').head()

SKU             Brand  \
10948  1000301P        Real Foods   
685      10016P   Victoria Bitter   
8915     10030P  Melbourne Bitter   
7052    100537P         Steradent   
659    1007071P          Babylove   

                                             Product Name Current Price  \
10948                             Tasty Cheese Corn Thins          1.80   
685                                             Can 375mL         48.00   
8915                                            Can 375mL         58.00   
7052                             Powder Denture Cleansing          5.95   
659    Unisex Stretchy Walker Nappy Pants 12-17 Kg Size 5         11.80   

       Price per unit Unit Size Previous Price On Special  \
10948  $1.44 per 100G      125g            2.0   $pecials   
685      $5.33 per 1L   24 Pack           53.0   $pecials   
8915     $6.44 per 1L   24 Pack                             
7052   $2.98 per 100G      200g                             
659     $0.47 per 1Ea   25 pack           17.0   $pecials   

                                     Suppliers             Category  \
10948                    ['731170 REAL FOODS']  Summer Entertaining   
685                           ['9393 CUB P/L']               Liquor   
8915                          ['9393 CUB P/L']               Liquor   
7052   ['672882 RECKITT BENCKISER (AUST) P/L']      Health & Beauty   
659            ['386332 UNICHARM AUSTRALASIA']                 Baby   

      Availability  \
10948    Available   
685      Available   
8915     Available   
7052     Available   
659      Available   

                                                                                         URL  
10948  https://shop.coles.com.au/a/richmond-south/product/real-foods-corn-thins-tasty-cheese  
685     https://shop.coles.com.au/a/richmond-south/product/victoria-bitter-375ml-cans-10016p  
8915           https://shop.coles.com.au/a/richmond-south/product/melbourne-bitter-can-375ml  
7052   https://shop.coles.com.au/a/richmond-south/product/steradent-denture-cleansing-powder  
659           https://shop.coles.com.au/a/richmond-south/product/babylove-nappy-pants-walker

Confirmed that SKUs indeed reference the exact same product.

Interestingly enough, we can already see that some products have a pretty significant price difference *e.g. A slab of Victoria Bitter has a \\$10 difference*. Other products seem to be priced the same.


One thing I noticed from reviewing other data, the "National" data (which can be obtained from the list of product URLs located on their sitemap) doesn't have any actual alcoholic products listed under the 'Liquor' category. Judging from this peculiarity, perhaps, the reason might be because those are the only products that differ in price across Australia??

It's also worth noting Coles technically doesn't sell alcohol (I've never seen alcohol stocked in any Coles physical stores). Perhaps these prices are from Liquorland?

## Product Availability Differences

* How many products are available on 26 January 2021, in Coles' online shop for Adelaide and Melbourne?
* How many products do they have in common?
* How many products are exclusive at each store on this particular day?

Finally: Output lists of Common Products and Exclusive Products of each location as CSV files.

#### How many products are available in Adelaide and Melbourne, respectively?

In [52]:
print(f'Adelaide has {len(df_adl)} available products')
print(f'Melbourne has {len(df_mel)} available products')

Adelaide has 19297 available products
Melbourne has 14409 available products


#### How many products in common?

In [53]:
# Shorten original dataframes to only have the necessary columns

adl_df = df_adl[['SKU', 'Brand', 'Product Name', 'Category', 'On Special', 'Current Price', 'Unit Size', 'Suppliers']]
adl_df = adl_df.rename(columns={
    'On Special': 'ADL Specials',
    'Current Price': 'ADL Price',
    'Unit Size': 'ADL packaging',
    'Suppliers': 'ADL Suppliers'
})

mel_df = df_mel[['SKU', 'Brand', 'Product Name', 'Category', 'On Special', 'Current Price', 'Unit Size', 'Suppliers']]
mel_df = mel_df.rename(columns={
    'On Special': 'MEL Specials',
    'Current Price': 'MEL Price',
    'Unit Size': 'MEL packaging',
    'Suppliers': 'MEL Suppliers'
})

In [54]:
# Inner Join of both dataframes and re-arrangements to juxtapose relevant columns

adl_mel_df = pd.merge(adl_df, mel_df, on=['SKU', 'Brand', 'Product Name', 'Category'], how='inner')

# Rename columns to reflect location attribute

adl_mel_df = adl_mel_df[['SKU', 'Brand', 'Product Name', 'Category',
                         'ADL Price', 'MEL Price',
                         'ADL packaging', 'MEL packaging',
                         'ADL Specials', 'MEL Specials',
                         'ADL Suppliers', 'MEL Suppliers']]

pd.set_option('display.max_rows', 200)
pd.set_option('display.max_colwidth', 200)
adl_mel_df.head(100)

SKU                       Brand  \
0   3468031P                       Coles   
1   2649234P                     Guylian   
2   8416240P                       Heinz   
3   2592829P                Jack Daniels   
4    324217P                       Coles   
5   5143435P                       Nivea   
6   3716508P                      Peters   
7   5430233P                Earth Choice   
8   3116216P                 Cook & Dine   
9   3671057P                    Grillman   
10  3815930P                     Tuckers   
11  2350546P                     Cadbury   
12  2511226P                Always Fresh   
13    93726P                      Massel   
14  3234154P                      Laucke   
15  3530292P                       Prima   
16  8211473P                       Coles   
17  3231098P                       Coles   
18  6446261P                       Radox   
19  5252500P                     Chaokoh   
20  1974116P                Pine O Cleen   
21  3558371P  The Wholesome Food Company   
22  3757549P                    Mainland   
23   126100P                  Campbell's   
24  3022484P                    Duracell   
25  1389364P                   Aeroplane   
26  5080812P                        Coon   
27  3492932P                       Coles   
28  3862930P                        Dole   
29  3623484P                        Baci   
30  3613130P                     Cadbury   
31  3797995P                   Fodmapped   
32  2351684P                     Cadbury   
33   171477P                     Sunbeam   
34  3449875P                  Life-Space   
35  3783822P                      Yo Pro   
36  7547898P                     Sunsilk   
37  5722374P                       Coles   
38  2680755P                       Coles   
39   399045P                       Hoyts   
40  3739048P                        Amoy   
41  3062922P                       Coles   
42  3279538P                Nature's Way   
43  3842692P                     Patties   
44  4854370P                     Sunrice   
45  2941156P                       Flora   
46  6893736P                     Streets   
47  2375274P                 TOM Organic   
48  6020850P                     Portone   
49  2858766P                    Hercules   
50  9189347P                Earth Choice   
51  4575514P                       Coles   
52   246102P                         Oxo   
53  8729551P                    Twinings   
54  2862515P                   Clearblue   
55  3516145P                 Wildly Good   
56  7767770P                      Nestle   
57  2044346P                   Mirabella   
58  9207427P                      V.I.P.   
59  3245354P                       Coles   
60  5060666P                    Twinings   
61  3773159P                        Mars   
62  2732065P                       Coles   
63  3303086P                     Garnier   
64  3209827P                   John West   
65  1987129P                 Planet Food   
66  1200118P                    Liddells   
67  7506981P                Conquistador   
68  8772700P                 Daily Juice   
69  9505163P                      Dilmah   
70  4575140P              Vegie Delights   
71  8370400P               Cobram Estate   
72  2360991P               Passage Foods   
73  9374531P                        Raid   
74  9051973P               Golden Circle   
75  7910660P                       Coles   
76  1898801P                       Decor   
77  4854380P                     Sunrice   
78  8145856P                       Primo   
79  9604631P                       Coles   
80  3580177P                         I&J   
81   181652P                        Bega   
82  3490030P                       Heinz   
83  4499085P                     Colgate   
84  3763541P                     Loacker   
85  3468790P                   Schweppes   
86  2768299P                  Mutti Sugo   
87  2904692P                        Nair   
88  3780278P                  Black Swan   
89  6636132P                  Go Natural   
90  4

In [55]:
print(f'There are {len(adl_mel_df)} items in common between Adelaide and Melbourne')

There are 12523 items in common between Adelaide and Melbourne


#### How many products are available in Adelaide but not in Melbourne?

In [100]:
# Anti merge by using the indicator tag to get the products exclusive to Adelaide (left dataframe)

df_outter_join = pd.merge(adl_df, mel_df, on=['SKU', 'Brand', 'Product Name', 'Category'], how="outer", indicator=True)

unique_adl_df = df_outter_join[['SKU', 'Brand', 'Product Name', 
                                'Category', 'ADL Price', 'ADL Suppliers']].loc[df_outter_join['_merge'] == 'left_only']
unique_adl_df.head(20)

SKU             Brand  \
0   3833501P    Farmers Market   
5   2015360P         1000 Hour   
9   3710829P          Tresemme   
10  3614596P              Tigi   
13  3830977P             Coles   
18  3339390P            Daawat   
21  9125050P             Kirks   
24  2709722P  Balfours Premium   
25  3836328P              Olay   
29  3761502P              Kynd   
30  8879758P           Beijing   
31  3712020P           L'Oreal   
32  2917355P           Nippy's   
35  3813934P            Revlon   
36  3844166P     Rey Del Mundo   
37  3824522P             Coles   
40  3731997P           St Ives   
41  8424317P           Kolotex   
47  3136224P       Cook & Dine   
52  3823428P            Myaura   

                                                   Product Name  \
0   Homestyle Shreds With Chicken Carrot & Cranberries Dog Food   
5                                  Black Eyelash & Brow Dye Kit   
9                               Violet Blonde Shine Conditioner   
10           Bed Head reatment Mask urban Antidotes L2 Recovery   
13                                      Seashore Royale Platter   
18                                   Everyday Basmati Rice Gold   
21                           Lemon Squash Multipack Cans 375 mL   
24                                              Peppersteak Pie   
25                                 Hydrating Caviar Lime Scrubs   
29                                       Glow Hair Skin & Nails   
30                                    Sliced Garlic Tofu Pieces   
31                     Volume Million Lashes Waterproof Mascara   
32                           Iced Coffee Flavoured Milk Chilled   
35                              Colorstay Foundation Sand Beige   
36                                    Organic Tempranillo 750mL   
37                              Gluten Free Rustic Grain Vienna   
40                            Energising Coconut & Coffee Scrub   
41            Sheer Relief Trouser Sock Beige One size Fits All   
47                                       Stainless Steel Peeler   
52                   Mandarin Grapefruit Lemon stick  Deodorant   

               Category ADL Price  \
0                   Pet      13.0   
5       Health & Beauty      20.0   
9       Health & Beauty      13.0   
10      Health & Beauty      20.9   
13        From The Deli      55.0   
18  International Foods      14.0   
21  Summer Entertaining       6.5   
24  Dairy, Eggs & Meals       4.2   
25      Health & Beauty      19.0   
29      Health & Beauty      30.0   
30  Dairy, Eggs & Meals       5.0   
31  Summer Entertaining      25.0   
32  Dairy, Eggs & Meals       2.0   
35      Health & Beauty      30.0   
36               Liquor      16.0   
37               Bakery       5.0   
40      Health & Beauty      10.0   
41             Clothing       4.0   
47            Household       4.5   
52      Health & Beauty       9.0   

                                                                                                             ADL Suppliers  
0                                                                              ['721182 VIP PETFOODS (AUST) P/L (CHILLED']  
5                                          ['145823 OVERSEAS SUPPLIER', '942368 CHEMCORP INTL', '969600 Chemcorp pty Ltd']  
9                                                                               ['26867 UNILEVER A/ASIA HOME &PERSONAL C']  
10                                                                              ['26867 UNILEVER A/ASIA HOME &PERSONAL C']  
13                                                                                     ['656186 COLES SUPERMARKETS (EXP)']  
18                                                                       ['970023 Sabi Foods', '972534 SABRINI FOODS P/L']  
21  ['931858 COCA COLA AMATIL AUST P/L (DIREC', '931859 COCA COLA AMATIL AUST P/L (GHPL)', '37079 LACTALIS AUSTRALIA P/L']  
24                                                                                          ['954304 BALFOUR

In [57]:
print(f"There are {len(unique_adl_df)} only available in Adelaide which are not available in Melbourne")

There are 6774 only available in Adelaide which are not available in Melbourne


#### Category breakdown of those products.

In [58]:
unique_adl_df['Category'].value_counts()

Health & Beauty         1799
Household                876
Pantry                   744
Dairy, Eggs & Meals      457
Pet                      388
Frozen                   371
Back to School           367
International Foods      310
Bakery                   296
Meat & Seafood           255
Summer Entertaining      197
Drinks                   186
Fruit & Vegetables       138
Liquor                   132
Baby                     111
From The Deli             64
                          31
Convenience Meals         28
Clothing                  22
New                        1
Chilled & Fresh Food       1
Name: Category, dtype: int64

#### How many products are available in Melbourne but not in Adelaide?

In [101]:
unique_mel_df = df_outter_join[['SKU', 'Brand', 'Product Name', 
                                'Category', 'MEL Price', 'MEL Suppliers']].loc[df_outter_join['_merge'] == 'right_only']
unique_mel_df.head(20)

SKU              Brand  \
19297  3387381P              Deeko   
19298  3312600P             Multix   
19299  3292886P           Femfresh   
19300    11549P          Budweiser   
19301  6100573P        Saint Cosme   
19302  3519711P                KOi   
19303  9759886P              Pauls   
19304  3911809P              Primo   
19305  3766459P  Rafferty's Garden   
19306  3114050P     Land of Giants   
19307  3915402P         White Claw   
19308  8371390P              Coles   
19309  3722362P              Coles   
19310  3331831P            Tibaldi   
19311  3534599P          Karicare+   
19312  3866533P       Lean Cuisine   
19313  3305070P              Coles   
19314  3454933P            Huggies   
19315  2523291P       Majans Bhuja   
19316  3882370P          Taokaenoi   

                                         Product Name             Category  \
19297                         Collection Dinner Plate  Summer Entertaining   
19298                   ReuseMe Resealable Snack Bags       Back to School   
19299  Pure & gentle For Sensitive Skin Intimate Care      Health & Beauty   
19300                                    Bottle 330mL               Liquor   
19301                            Cotes du Rhone 750mL               Liquor   
19302         Arabica Coffee & Shea Butter Body Scrub      Health & Beauty   
19303                              Smarter White Milk  Dairy, Eggs & Meals   
19304                            Shaved Champange Ham       Back to School   
19305                      Summer Fruits & Quinoa 6M+                 Baby   
19306                                Pinot Gris 750mL               Liquor   
19307                        Hard Seltzer Mango 330mL  Summer Entertaining   
19308                Traditional Fruit Hot Cross Buns               Bakery   
19309                                Unicorn Crumpets               Bakery   
19310                                   Turkey Breast       Back to School   
19311                     Goat Stage 1 Infant Formula                 Baby   
19312                           Indian Butter Chicken               Frozen   
19313                                    Lamb Cutlets       Meat & Seafood   
19314                   Fragrance Free 400 Baby Wipes                 Baby   
19315                            Gluten Free Beer Mix  Summer Entertaining   
19316                         Classic Big Roll 9 Pack  International Foods   

      MEL Price  \
19297      5.00   
19298      2.40   
19299      6.75   
19300     19.00   
19301     22.00   
19302      5.60   
19303      5.55   
19304      7.00   
19305      1.90   
19306     17.00   
19307     24.00   
19308      3.50   
19309      4.00   
19310      3.00   
19311     40.00   
19312      6.70   
19313     13.44   
19314     13.00   
19315      2.00   
19316      5.00   

                                                                                                                                                                                                 MEL Suppliers  
19297                                                                                                                                                                            ['4545 THE SORBENT PAPER CO']  
19298                                                                                                                                                                ["8125 MCPHERSON'S CONSUMER PRODUCTS P/"]  
19299                                                                                                                                                                      ['7641 CHURCH & DWIGHT (AUST) P/L']  
19300  ['169871 CASTLEMAINE PERKINS LTD', '42196 THE SWAN BREWERY CO P/L', '975969 LION-BEER SPIRITS & WINE PTY LTD', '729248 AUSTRALIAN LIQUOR MARKETERS P/L', '710911 LION NATHAN VICTORIA P/L (BEER)', '...  
19301                                                                                                                            

In [60]:
print(f"There are {len(unique_mel_df)} products unique to Melbourne")

There are 1886 products unique to Melbourne


#### Category Breakdown

In [61]:
unique_mel_df['Category'].value_counts()

Liquor                 509
Pantry                 357
Health & Beauty        195
Dairy, Eggs & Meals    114
Meat & Seafood          87
Summer Entertaining     85
Back to School          85
Household               83
Baby                    77
Drinks                  65
International Foods     60
Frozen                  54
Pet                     33
Fruit & Vegetables      32
Bakery                  31
                        13
From The Deli            3
Convenience Meals        3
Name: Category, dtype: int64

#### Outputting DataFrames to CSV

In [62]:
# Output all the above dataframes to CSV for better viewing in Excel

# Common Products List
adl_mel_df.to_csv('Outputs/Common Products ADL-MEL.csv', index=False, encoding='utf-8-sig')

# Unique to Adelaide
unique_adl_df.to_csv('Outputs/Unique to ADL.csv', index=False, encoding='utf-8-sig')

# Unique to Melbourne
unique_mel_df.to_csv('Outputs/Unique to MEL.csv', index=False, encoding='utf-8-sig')

#### Summary Insights:

* Adelaide had __19,297__ available products. Melbourne had __14,409__ available products
* There were __12,523__ items in common
* Adelaide had __6774__ products which were unavailable in the Melbourne online store. The majority of these products were in the _'Health & Beauty', 'Household'_ and _'Pantry'_ categories.
* Melbourne had __1886__ products which were unavailable in the Adelaide online store. The majority of these products were in the _'Liquor', 'Pantry'_ and _'Health & Beauty'_ categories.

<mark>See the three CSV file dumps in the 'Outputs' folder for further details.</mark>

<br>

## Suppliers

* Are all the suppliers the same for all of the products in common?
* Who are the suppliers of the products only available in Adelaide? (and vice versa)
* Are there any products with different suppliers?

#### Are all the suppliers the same for all the common products? (most likely, yes, but i'm curious)

In [63]:
# Out of curiosity... are the suppliers all the same?

filter = adl_mel_df['ADL Suppliers'] != adl_mel_df['MEL Suppliers']

different_suppliers_df = adl_mel_df.loc[filter]

print(f"Products with different suppliers: {len(different_suppliers_df)}")

Products with different suppliers: 0


#### Suppliers of Adelaide unique products not listed as a supplier in the Melbourne store

In [64]:
filter = unique_adl_df['ADL Suppliers'].isin(df_mel['Suppliers'])

unique_adl_df[~filter][['Brand', 'Product Name', 'Category', 'ADL Suppliers']]

Brand                         Product Name  \
5                1000 Hour         Black Eyelash & Brow Dye Kit   
18                  Daawat           Everyday Basmati Rice Gold   
21                   Kirks   Lemon Squash Multipack Cans 375 mL   
24        Balfours Premium                      Peppersteak Pie   
30                 Beijing            Sliced Garlic Tofu Pieces   
...                    ...                                  ...   
19208                Coles               Brussels Sprouts loose   
19215  Ashton Valley Crush               Full Bodied Pear Juice   
19224                Ivans                        Apple Crumble   
19254           Maybelline  Matte Maker Natural Beige 30 Powder   
19272                Coles            Organic Carrots Prepacked   

                  Category  \
5          Health & Beauty   
18     International Foods   
21     Summer Entertaining   
24     Dairy, Eggs & Meals   
30     Dairy, Eggs & Meals   
...                    ...   
19208   Fruit & Vegetables   
19215               Drinks   
19224               Bakery   
19254      Health & Beauty   
19272   Fruit & Vegetables   

                                                                                                                                                                                                 ADL Suppliers  
5                                                                                                                              ['145823 OVERSEAS SUPPLIER', '942368 CHEMCORP INTL', '969600 Chemcorp pty Ltd']  
18                                                                                                                                                           ['970023 Sabi Foods', '972534 SABRINI FOODS P/L']  
21                                                                                      ['931858 COCA COLA AMATIL AUST P/L (DIREC', '931859 COCA COLA AMATIL AUST P/L (GHPL)', '37079 LACTALIS AUSTRALIA P/L']  
24                                                                                                                                                                              ['954304 BALFOURS BAKERY P/L']  
30                                                                                                                                                                   ['935403 BEIJING SOYA BEAN PRODUCTS P/L']  
...                                                                                                                                                                                                        ...  
19208  ['414760 ST GEORGE PROD PTY LTD', '939312 FRESH PRODUCE DC (NSW)', '942633 CARTER & SPENCER NORTH QLD', '97446 CARTER & SPENCER GROUP P/L', "635235 COSTA'S P/L (VIC DC)", '933347 COSTAS (TASMANIA)...  
19215                                                                                                           ['942529 R CERAVOLO & CO P/L', '983253 CERAVOLO ORCHARDS', '966748 CERAVOLO ORCHARDS PTY LTD']  
19224                                                                                                                                                                    ['945657 IVANS PIES (PRIVATE LABEL)']  
19254                                                                                                                                          ["720488 L'OREAL AUST P/L", "952096 L'OREAL AUST P/L (X-DOCK)"]  
19272  ['933347 COSTAS (TASMANIA)', '414760 ST GEORGE PROD PTY LTD', "635235 COSTA'S P/L (VIC DC)", '939312 FRESH PRODUCE DC (NSW)', '97446 CARTER & SPENCER GROUP P/L', '942633 CARTER & SPENCER NORTH QLD...  

[2172 rows x 4 columns]

#### Suppliers of Melbourne unique products not listed as a supplier in the Adelaide store

In [65]:
filter = unique_mel_df['MEL Suppliers'].isin(df_adl['Suppliers'])

unique_mel_df[~filter][['Brand', 'Product Name', 'Category', 'MEL Suppliers']]

Brand                                      Product Name  \
19300      Budweiser                                      Bottle 330mL   
19303          Pauls                                Smarter White Milk   
19308          Coles                  Traditional Fruit Hot Cross Buns   
19310        Tibaldi                                     Turkey Breast   
19323  Breathe Right             Clear Regular Nasal Congestion Strips   
...              ...                                               ...   
21157          Coles  Australian Peri Peri Roast Chicken Breast Slices   
21161           Dare                              Espresso Iced Coffee   
21163       Caprilac                            Pasteurised Goats Milk   
21174       Aqualove                                    Alkaline Water   
21177          Coles                                   Calypso Mangoes   

                  Category  \
19300               Liquor   
19303  Dairy, Eggs & Meals   
19308               Bakery   
19310       Back to School   
19323      Health & Beauty   
...                    ...   
21157  Dairy, Eggs & Meals   
21161  Dairy, Eggs & Meals   
21163  Dairy, Eggs & Meals   
21174       Back to School   
21177       Back to School   

                                                                                                                                                                                                 MEL Suppliers  
19300  ['169871 CASTLEMAINE PERKINS LTD', '42196 THE SWAN BREWERY CO P/L', '975969 LION-BEER SPIRITS & WINE PTY LTD', '729248 AUSTRALIAN LIQUOR MARKETERS P/L', '710911 LION NATHAN VICTORIA P/L (BEER)', '...  
19303                        ['944347 LACTALIS AUSTRALIA P/L', '37079 LACTALIS AUSTRALIA P/L', '964585 LACTALIS AUSTRALIA P/L', '964518 PARMALAT FOOD PRODUCTS P/L (NSW', '959594 PARMALAT FOOD PRODUCTS P/L']  
19308  ['29564 CRIPPS NUBAKE', '652199 TIP TOP', '953879 ALLIED MILLS P/L', '972720 ALLIED PINNACLE', '920925 TIP TOP IN-STORE BAKERY PRODUCTS', '937127 CRIPPS NUBAKE P/L', '965447 DAVIES BAKERY P/L (HOU...  
19310                                                                                                                            ['974006 TIBALDI AUSTRALASIA(FIXED WEIGHT', '968633 TIBALDI A/ASIA P/L (DY)']  
19323                                                                                       ['617857 MARKETING BROKERS AUST P/L', '971903 GLAXOSMITHKLINE CONSUMER HEALTHC', '41629 GLAXOSMITHKLINE AUST P/L']  
...                                                                                                                                                                                                        ...  
21157                                                                                                                                ['922163 MILNE AGRIGROUP P/L', '970470 THE CORDINA FOOD CO (COLES BRAND']  
21161                                                                                             ['966403 LD&D AUSTRALIA PTY LTD (DSD)', '367702 CAPITOL CHILLED FOODS (AUST) P/L', '665282 LD & D AUST P/L']  
21163                                                                                                                                 ['969734 THE INFANT FOOD CO. PTY LIMITED', '973144 NULAC FOODS PTY LTD']  
21174                                                                                                                                                    ['974947 AQUALOVE WATER', '145823 OVERSEAS SUPPLIER']  
21177  ['939312 FRESH PRODUCE DC (NSW)', '97446 CARTER & SPENCER GROUP P/L', "635235 COSTA'S P/L (VIC DC)", '414760 ST GEORGE PROD PTY LTD', '942633 CARTER & SPENCER NORTH QLD', '933347 COSTAS (TASMANIA)...  

[400 rows x 4 columns]

In [66]:
mel_suppliers_df = pd.read_csv('sauce/Jan 26 Melbourne VIC.csv')[
    ['SKU', 'Brand', 'Product Name', 'Suppliers']].rename(columns={'Suppliers': 'MEL Suppliers'})

adl_suppliers_df = pd.read_csv('sauce/Jan 26 Adelaide SA.csv')[
    ['SKU', 'Brand', 'Product Name', 'Suppliers']].rename(columns={'Suppliers': 'ADL Suppliers'})

In [67]:
# Suppliers found in Melbourne but not in Adelaide

filter = mel_suppliers_df['MEL Suppliers'].isin(adl_suppliers_df['ADL Suppliers'])

df_MelbSuppliersNotListedInADL = mel_suppliers_df[~filter]
df_MelbSuppliersNotListedInADL

SKU                     Brand                      Product Name  \
58     9759886P                     Pauls                Smarter White Milk   
99     3939706P                  Breadtop          Pork Floss Mayo Sandwich   
128    8371390P                     Coles  Traditional Fruit Hot Cross Buns   
145    3331831P                   Tibaldi                     Turkey Breast   
263    3780675P  The Organic Milk Company                     Salted Butter   
...         ...                       ...                               ...   
15947  3034010P                      Dare              Espresso Iced Coffee   
15974  3987232P                  Caprilac            Pasteurised Goats Milk   
15999  3112678P               Cook & Dine               Wooden Pastry Brush   
16044  3250300P                  Aqualove                    Alkaline Water   
16046  7384170P               Club Armada                   White Rum 700mL   

                                                                                                                                                                                                 MEL Suppliers  
58                           ['944347 LACTALIS AUSTRALIA P/L', '37079 LACTALIS AUSTRALIA P/L', '964585 LACTALIS AUSTRALIA P/L', '964518 PARMALAT FOOD PRODUCTS P/L (NSW', '959594 PARMALAT FOOD PRODUCTS P/L']  
99                                                                                                                                                                                         ['977923 BREADTOP']  
128    ['29564 CRIPPS NUBAKE', '652199 TIP TOP', '953879 ALLIED MILLS P/L', '972720 ALLIED PINNACLE', '920925 TIP TOP IN-STORE BAKERY PRODUCTS', '937127 CRIPPS NUBAKE P/L', '965447 DAVIES BAKERY P/L (HOU...  
145                                                                                                                              ['974006 TIBALDI AUSTRALASIA(FIXED WEIGHT', '968633 TIBALDI A/ASIA P/L (DY)']  
263                                                                                                                                                                        ['976052 TOMC BRANDS PTY LTD (CB)']  
...                                                                                                                                                                                                        ...  
15947                                                                                             ['966403 LD&D AUSTRALIA PTY LTD (DSD)', '367702 CAPITOL CHILLED FOODS (AUST) P/L', '665282 LD & D AUST P/L']  
15974                                                                                                                                 ['969734 THE INFANT FOOD CO. PTY LIMITED', '973144 NULAC FOODS PTY LTD']  
15999                                                                            ['970149 Globaleyes Distribution (COLESBR', '971613 GROUPE SEB AUSTRALIA PTY LTD', '945707 GLOBALEYES DISTRIBUTION P/L (XDO']  
16044                                                                                                                                                    ['974947 AQUALOVE WATER', '145823 OVERSEAS SUPPLIER']  
16046                                                                                                                             ['806543 FERNBREW P/L', '963979 MAX LIQUOR SUPPLIES', '954095 FERNBREW P/L']  

[327 rows x 4 columns]

#### Are there any products with different suppliers

In [94]:
'''
Are there any products in the list above (Suppliers not found in Adelaide) that can be found in Adelaide 
i.e. same product, different supplier
'''

filter = adl_suppliers_df['SKU'].isin(df_MelbSuppliersNotListedInADL['SKU'])
adl_suppliers_df[filter]
print(f'Total number of products in Adelaide with different suppliers to Melbourne: {len(adl_suppliers_df[filter])}')

Total number of products in Adelaide with different suppliers to Melbourne: 0


#### Inferences Drawn

Initially, it seemed like there were quite a few number of suppliers supplying one city and not the other. The issue with the initial outputs was the fact that I had filtered out the "unavailable" products right at the start. After opening up the CSV files in Excel and viewing them, it turns out some of the products e.g. Budweiser does have the exact same list of suppliers but was listed as "unavailable" in Adelaide.

So, I reloaded the CSV files into new dataframes, this time including all of the "unavailable" products. The results were pretty much as expected. The suppliers not found in the Melbourne stores were for all of the products which were also "unavailable" or not listed in the Melbourne store. It's highly likely that to be a supplier for Coles, one must have the capability to supply Nationally. However, since Coles lists a list of suppliers, it's possible that some of them only supply exclusively to one area.

<br>

## Melbourne vs Adelaide - Pricing Differences


* What products have the same price? What categories are most prevelant?
* What are the identical products with different prices?
* Are all the same products on "special" between these two cities?
* Are pricing differences due to only one store having that product on "special"?
* Are there any trends in these pricing differences i.e. are the pricing differences heavily within one category or is it influenced by brand?

In [69]:
# Create new column 'price difference' ==> the absolute value of (Adelaide - Melbourne)

adl_mel_df.insert(6, 'Price Difference', abs(adl_mel_df['ADL Price'] - adl_mel_df['MEL Price']))

# Create new column 'Specials Comp' which compares if '$pecial' tag are matching across two stores

adl_mel_df.insert(11, 'Same Specials', adl_mel_df['ADL Specials'] == adl_mel_df['MEL Specials'])

In [103]:
# Let's have a peek at top 5 price differences

adl_mel_df.sort_values('Price Difference', ascending=False).head(20)

SKU            Brand                               Product Name  \
10725  4298877P          Coopers                      Mild Ale Bottle 375mL   
6906    479615P            Ruski                         Lemon Bottle 275mL   
12350    10078P  Victoria Bitter                               Bottle 375mL   
11896  1274330P    Canadian Club                         & Dry Bottle 500mL   
11562    10016P  Victoria Bitter                                  Can 375mL   
388    2123668P            Coles                   Roasted & Salted Cashews   
3455   1263016P         Somersby                       Pear Cider Can 375mL   
8254    539470P         Smirnoff                      Ice Red Bottles 300mL   
8235   3207774P          Coopers                      Pacific Ale Can 375mL   
1175   2821802P   Captain Morgan                     Cola Bottle 4.5% 330mL   
7201   1253180P         Somersby                      Apple Cider Can 375mL   
1046   2554003P   Hammer N Tongs                     Bitter Block Can 330mL   
3963   3479333P  4 Pines Brewing                   Pacific Ale Bottle 330mL   
3738   1621794P    Mountain Goat                       Summer Ale Can 375mL   
8364   3369222P          Coopers                           Dry Bottle 355mL   
6535   3223908P           Corona                        Ligera Bottle 355mL   
4424   3663037P      Pirate Life               Throwback IPA 3.5% Can 355mL   
2096   6577554P             XXXX                       Gold Block Can 375mL   
5530     36191P            Crown                         Lager Bottle 375mL   
11897  3613108P     Mountain Dew  Energised Multipack Soft Drink Cans 375mL   

                 Category ADL Price MEL Price Price Difference ADL packaging  \
10725              Liquor     44.00      16.0            28.00        6 Pack   
6906               Liquor    107.00      91.0            16.00       24 Pack   
12350              Liquor     58.00      47.0            11.00       24 Pack   
11896              Liquor     90.00      79.0            11.00       12 Pack   
11562              Liquor     58.00      48.0            10.00       24 Pack   
388    Fruit & Vegetables     10.00      16.0             6.00          800g   
3455               Liquor     72.00      66.0             6.00       30 Pack   
8254               Liquor    105.00     100.0             5.00       24 Pack   
8235               Liquor     58.00      63.0             5.00       24 Pack   
1175               Liquor    116.00     111.0             5.00       24 pack   
7201               Liquor     71.00      66.0             5.00       30 Pack   
1046               Liquor     44.00      39.0             5.00       30 Pack   
3963               Liquor     61.00      56.0             5.00       24 Pack   
3738               Liquor     24.50      20.0             4.50        6 Pack   
8364               Liquor     50.00      46.0             4.00       24 Pack   
6535               Liquor     50.00      46.0             4.00       24 Pack   
4424               Liquor     50.00      46.0             4.00       16 Pack   
2096               Liquor     47.00      43.0             4.00       30 Pack   
5530               Liquor     59.00      55.0             4.00       24 Pack   
11897              Drinks     14.15      10.5             3.65       24 Pack   

      MEL packaging ADL Specials MEL Specials  Same Specials  \
10725        6 Pack                                     True   
6906        24 Pack                                     True   
12350       24 Pack                  $pecials          False   
11896       12 Pack                                     True   
11562       24 Pack                  $pecials          False   
388            800g                                     True   
3455        30 Pack                                     True   
8254        24 Pack                                     True   
8235        24 Pack                                     True   
1175        24 pack            

#### What products have the same price in both Adelaide and Melbourne?

In [102]:
# ADL vs MEL same price'

same_price_df = adl_mel_df.loc[adl_mel_df['Price Difference'] == 0]

same_price_df.head(20)

SKU         Brand                              Product Name  \
0   3468031P         Coles     Natures Kitchen Lentil Spag Bolognese   
1   2649234P       Guylian             Assortment Belgian Chocolates   
2   8416240P         Heinz                   Big Red Tomato Soup Can   
4    324217P         Coles                      Smart Buy Hair Spray   
5   5143435P         Nivea   Moisturising Day Face Cream Fluid Spf15   
6   3716508P        Peters          Connoisseur Strawberry Ice Cream   
7   5430233P  Earth Choice                 Pure Soft Fabric Softener   
8   3116216P   Cook & Dine                                 Corkscrew   
9   3671057P      Grillman        BBQ Aluminium Large Roasting Trays   
10  3815930P       Tuckers           Natural Artisan Crackers Seeded   
11  2350546P       Cadbury  Old Gold Peppermint Dark Chocolate Block   
12  2511226P  Always Fresh               Flat Anchovies In Olive Oil   
13    93726P        Massel                    Beef Style Stock Cubes   
14  3234154P        Laucke                             CWA Scone Mix   
15  3530292P         Prima      Tropical Fruit Drink Multipack 200mL   
16  8211473P         Coles                             Sunflower Oil   
17  3231098P         Coles                      5 Brown Rice Pouches   
18  6446261P         Radox                          Oxygen Body Wash   
19  5252500P       Chaokoh                          Young Sweet Corn   
20  1974116P  Pine O Cleen   Multi Purpose Trigger Spray Crisp Apple   

               Category ADL Price MEL Price Price Difference ADL packaging  \
0   Dairy, Eggs & Meals      7.00      7.00              0.0          350g   
1                Pantry     18.00     18.00              0.0          305g   
2                Pantry      3.65      3.65              0.0          820g   
4       Health & Beauty      2.20      2.20              0.0          250g   
5       Health & Beauty      5.35      5.35              0.0          50mL   
6                Frozen     11.00     11.00              0.0            1L   
7             Household      2.00      2.00              0.0            1L   
8             Household      3.00      3.00              0.0        1 each   
9             Household      4.80      4.80              0.0        2 pack   
10               Pantry      3.20      3.20              0.0          100g   
11               Pantry      5.00      5.00              0.0          180g   
12               Pantry      4.50      4.50              0.0          100g   
13               Pantry      2.50      2.50              0.0          105g   
14               Pantry      5.15      5.15              0.0           1kg   
15       Back to School      2.50      2.50              0.0        6 pack   
16               Pantry      9.75      9.75              0.0            4L   
17               Pantry      4.00      4.00              0.0          625g   
18      Health & Beauty      2.00      2.00              0.0          50mL   
19  International Foods      2.00      2.00              0.0          425g   
20            Household      3.50      3.50              0.0         750mL   

   MEL packaging ADL Specials MEL Specials  Same Specials  \
0           350g     $pecials     $pecials           True   
1           305g                                     True   
2           820g                                     True   
4           250g                                     True   
5           50mL     $pecials     $pecials           True   
6             1L                                     True   
7             1L                                     True   
8         1 each                                     True   
9         2 pack     $pecials     $pecials           True   
10          100g     $pecials     $pecials           True   
11          180g                                     True   
12          100g     $pecials     $pecials           True   
13          105g     $pecials     $pecials           True   
14 

#### Category breakdown of products with identical prices

In [72]:
print(f"Total number of common products between ADL and MEL: {len(same_price_df)} \n")

print(f"{same_price_df.value_counts('Category')}")

Total number of common products between ADL and MEL: 11811 

Category
Pantry                 3985
Health & Beauty        1674
Back to School         1002
Household               966
Summer Entertaining     721
Dairy, Eggs & Meals     692
Liquor                  612
Frozen                  562
Baby                    345
Pet                     310
International Foods     256
Drinks                  235
Bakery                  146
Meat & Seafood          121
Fruit & Vegetables       98
                         53
From The Deli            18
Convenience Meals        10
New                       4
Clothing                  1
dtype: int64


#### What are the products that are not 'on special' at the same time but have matching prices

In [73]:
same_price_df.loc[same_price_df['Same Specials'] == False]

SKU           Brand  \
2286   2860768P           Fairy   
2552   2289015P           Heinz   
3259   2374191P  Great Northern   
4962   3780369P  Vegie Delights   
5110   1163725P           Heinz   
5423   1163689P           Heinz   
7674   2752255P           Coles   
9756   3663060P     Pirate Life   
10000  2244527P           Coles   

                                             Product Name  \
2286         Original All In One Lemon Dishwasher Tablets   
2552   Soup Of The Day 7 Veg With Garden Herbs Soup Pouch   
3259                       Super Crisp Lager Bottle 330mL   
4962                                      Bolognese Sauce   
5110          Soup of The Day Old Fashioned Chicken Pouch   
5423              Soup Of The Day Buttercup Pumpkin & Veg   
7674                           Nature's Kitchen Firm Tofu   
9756                              Pale Ale 5.4% Can 355mL   
10000          Ultra Complete Clean Dishwasher 30 Tablets   

                  Category ADL Price MEL Price Price Difference ADL packaging  \
2286             Household      19.0      19.0              0.0       44 pack   
2552                Pantry       3.0       3.0              0.0          430g   
3259                Liquor      18.0      18.0              0.0        6 Pack   
4962                Pantry       5.0       5.0              0.0          300g   
5110                Pantry       3.0       3.0              0.0          430g   
5423                Pantry       3.0       3.0              0.0          430g   
7674   Dairy, Eggs & Meals       2.3       2.3              0.0          300g   
9756                Liquor      59.0      59.0              0.0       16 Pack   
10000            Household       4.5       4.5              0.0          540g   

      MEL packaging ADL Specials MEL Specials  Same Specials  \
2286        44 pack                  $pecials          False   
2552           430g                  $pecials          False   
3259         6 Pack     $pecials                       False   
4962           300g     $pecials                       False   
5110           430g                  $pecials          False   
5423           430g                  $pecials          False   
7674           300g     $pecials                       False   
9756        16 Pack     $pecials                       False   
10000          540g     $pecials                       False   

                                                                 ADL Suppliers  \
2286                                      ['422720 PROCTER & GAMBLE AUST P/L']   
2552                                            ['22161 HJ HEINZ CO AUST LTD']   
3259                                                          ['9393 CUB P/L']   
4962                                              ['971159 LIFE HEALTH FOODS']   
5110                                            ['22161 HJ HEINZ CO AUST LTD']   
5423                                            ['22161 HJ HEINZ CO AUST LTD']   
7674                                ['972547 AUSTRALIAN EATWELL PTY LTD (CB)']   
9756                                            ['963979 MAX LIQUOR SUPPLIES']   
10000  ['971094 MCBRIDE AUSTRALIA PTY LTD (COLES', '145823 OVERSEAS SUPPLIER']   

                                                                 MEL Suppliers  
2286                                      ['422720 PROCTER & GAMBLE AUST P/L']  
2552                                            ['22161 HJ HEINZ CO AUST LTD']  
3259                                                          ['9393 CUB P/L']  
4962                                              ['971159 LIFE HEALTH FOODS']  
5110                                            ['22161 HJ HEINZ CO AUST LTD']  
5423                                            ['22161 HJ HEINZ CO AUST LTD']  
7674                                ['972547 AUSTRALIAN EATWELL PTY LTD (CB)']  
9756                                            ['963979 MAX LIQUOR SUPPLIES']  
10000  ['971094 MCBRIDE AUSTRALIA PTY LTD (COLES

#### Summary of products with same prices

In [93]:
print(f'Total number of products common to both Cole\'s Online in Adelaide and Melbourne stores: {len(adl_mel_df)}\n')
print(f'There are {len(same_price_df)} identical items with identical prices\n')
print(f'There are {len(same_price_df.loc[same_price_df["Same Specials"] == False])} items which have different \"on specials\" status but have the same price\n')
print(f'{round(100*len(same_price_df)/len(adl_mel_df),2)}% of the products common between Adelaide and Melbourne have identical prices\n')

Total number of products common to both Cole's Online in Adelaide and Melbourne stores: 12523

There are 11811 identical items with identical prices

There are 9 items which have different "on specials" status but have the same price

94.31% of the products common between Adelaide and Melbourne have identical prices



#### What are the products with different prices?

In [104]:
diff_price_df = adl_mel_df.loc[adl_mel_df['Price Difference'] != 0]

diff_price_df.sort_values('Price Difference', ascending=False).head(20)

SKU            Brand                     Product Name  \
10725  4298877P          Coopers            Mild Ale Bottle 375mL   
6906    479615P            Ruski               Lemon Bottle 275mL   
11896  1274330P    Canadian Club               & Dry Bottle 500mL   
12350    10078P  Victoria Bitter                     Bottle 375mL   
11562    10016P  Victoria Bitter                        Can 375mL   
388    2123668P            Coles         Roasted & Salted Cashews   
3455   1263016P         Somersby             Pear Cider Can 375mL   
8235   3207774P          Coopers            Pacific Ale Can 375mL   
3963   3479333P  4 Pines Brewing         Pacific Ale Bottle 330mL   
1046   2554003P   Hammer N Tongs           Bitter Block Can 330mL   
7201   1253180P         Somersby            Apple Cider Can 375mL   
1175   2821802P   Captain Morgan           Cola Bottle 4.5% 330mL   
8254    539470P         Smirnoff            Ice Red Bottles 300mL   
3738   1621794P    Mountain Goat             Summer Ale Can 375mL   
8364   3369222P          Coopers                 Dry Bottle 355mL   
6535   3223908P           Corona              Ligera Bottle 355mL   
2096   6577554P             XXXX             Gold Block Can 375mL   
4424   3663037P      Pirate Life     Throwback IPA 3.5% Can 355mL   
5530     36191P            Crown               Lager Bottle 375mL   
10809  9391574P        Schweppes  Solo Lemon Multipack Cans 375mL   

                  Category ADL Price MEL Price Price Difference ADL packaging  \
10725               Liquor     44.00      16.0            28.00        6 Pack   
6906                Liquor    107.00      91.0            16.00       24 Pack   
11896               Liquor     90.00      79.0            11.00       12 Pack   
12350               Liquor     58.00      47.0            11.00       24 Pack   
11562               Liquor     58.00      48.0            10.00       24 Pack   
388     Fruit & Vegetables     10.00      16.0             6.00          800g   
3455                Liquor     72.00      66.0             6.00       30 Pack   
8235                Liquor     58.00      63.0             5.00       24 Pack   
3963                Liquor     61.00      56.0             5.00       24 Pack   
1046                Liquor     44.00      39.0             5.00       30 Pack   
7201                Liquor     71.00      66.0             5.00       30 Pack   
1175                Liquor    116.00     111.0             5.00       24 pack   
8254                Liquor    105.00     100.0             5.00       24 Pack   
3738                Liquor     24.50      20.0             4.50        6 Pack   
8364                Liquor     50.00      46.0             4.00       24 Pack   
6535                Liquor     50.00      46.0             4.00       24 Pack   
2096                Liquor     47.00      43.0             4.00       30 Pack   
4424                Liquor     50.00      46.0             4.00       16 Pack   
5530                Liquor     59.00      55.0             4.00       24 Pack   
10809  Summer Entertaining     14.15      10.5             3.65       24 pack   

      MEL packaging ADL Specials MEL Specials  Same Specials  \
10725        6 Pack                                     True   
6906        24 Pack                                     True   
11896       12 Pack                                     True   
12350       24 Pack                  $pecials          False   
11562       24 Pack                  $pecials          False   
388            800g                                     True   
3455        30 Pack                                     True   
8235        24 Pack                                     True   
3963        24 Pack     $pecials     $pecials           True   
1046        30 Pack                                     True   
7201        30 Pack                                     True   
1175        24 pack                                     True   
8254        24 Pack         

In [76]:
# Output to CSV

diff_price_df.to_csv('Outputs/Different Priced Products ADL-MEL 26Jan21.csv', index=False, encoding='utf-8-sig')

In [77]:
print(f'There are {len(diff_price_df)} items with different prices')

There are 712 items with different prices


#### After filtering out products which are not 'on special' simultaneously

In [78]:
# Dataframe after accounting for 'on special' status differences

diff_price_spec_adj_df = diff_price_df.loc[diff_price_df['Same Specials'] == True]

In [79]:
print(f'After adjusting for differences in the "on special" status, there are {len(diff_price_spec_adj_df)} items')

After adjusting for differences in the "on special" status, there are 690 items


In [80]:
print(diff_price_spec_adj_df['Category'].value_counts(normalize=True)*100)

print(f"The top 3 Categories account for {100 * sum(diff_price_spec_adj_df['Category'].value_counts(normalize=True)[:3])}%") 

Liquor                 25.362319
Drinks                 23.768116
Summer Entertaining    22.608696
Pantry                  7.971014
Back to School          7.246377
Fruit & Vegetables      4.492754
Health & Beauty         2.608696
Household               2.028986
Dairy, Eggs & Meals     1.594203
International Foods     0.724638
                        0.434783
Baby                    0.434783
Pet                     0.434783
Frozen                  0.289855
Name: Category, dtype: float64
The top 3 Categories account for 71.73913043478262%


#### Brands most prevelant in products with price differences

In [81]:
print(diff_price_spec_adj_df['Brand'].value_counts(normalize=True).loc[lambda x: x>0.01]*100)

print(f"{len(diff_price_spec_adj_df['Brand'].value_counts(normalize=True).loc[lambda x: x>0.01])} Brands that account for more than 1% of the total, each.")

print(f"These 20 brands account for {sum(diff_price_spec_adj_df['Brand'].value_counts(normalize=True).loc[lambda x: x>0.01]) * 100}% of the list total")

Coles             15.652174
Schweppes          5.942029
Coca-Cola          3.913043
Bundaberg          2.463768
Pepsi              2.318841
Mount Franklin     2.173913
Lipton             1.739130
Red Bull           1.739130
Jack Daniels       1.594203
Kirks              1.449275
MasterFoods        1.304348
Smirnoff           1.304348
Coopers            1.304348
Somersby           1.159420
Powerade           1.159420
Nexba              1.159420
Vodka              1.159420
Madura             1.014493
Monster            1.014493
Pure Blonde        1.014493
Name: Brand, dtype: float64
20 Brands that account for more than 1% of the total, each.
These 20 brands account for 50.57971014492754% of the list total


#### Most common brands with price differences, by category

In [82]:
diff_price_spec_adj_df.groupby('Category')['Brand'].value_counts().loc[lambda x : x>4]

Category             Brand         
Back to School       Coles             26
                     Mount Franklin     5
Drinks               Coles             16
                     Red Bull          12
                     Coca-Cola         11
                     Schweppes         11
                     Mount Franklin    10
                     Bundaberg          8
                     Lipton             8
                     Powerade           8
                     Monster            7
                     Nexba              5
                     Pepsi              5
                     V                  5
Fruit & Vegetables   Coles             31
Liquor               Jack Daniels      10
                     Coopers            8
                     Smirnoff           7
                     Jim Beam           6
                     Pure Blonde        6
                     Somersby           6
                     Captain Morgan     5
                     Woodstock          

#### Inferences Drawn



In [105]:
# What if we exclude all the alcoholic and non-alcoholic beverages?

filter = (
    diff_price_spec_adj_df['Category'].str.contains('Liquor|Drinks') |
    diff_price_spec_adj_df['Product Name'].str.contains('mL|Drink|Water|Cola|Beer|Soda') |
    diff_price_spec_adj_df['Brand'].isin(['Schweppes', 'Coca-Cola', 'Pepsi', 'Mountain Dew',
                                          'Lipton', 'Mount Franklin', 'Fanta', 'Sunkist'])
)

nonbev_diff_price_df = diff_price_spec_adj_df.loc[~filter]
nonbev_diff_price_df.sort_values('Price Difference', ascending=False).head(20)

SKU       Brand  \
388    2123668P       Coles   
4364   2123602P       Coles   
8711   2743866P       Coles   
490    4879042P       Coles   
5140   3866840P       Coles   
9531   4845732P       Coles   
2731   9932765P  Lifestyles   
9407   4456832P       Coles   
12442  8900989P     Aptamil   
6366   6763800P     Aptamil   
3003   2131611P       Coles   
7534   4619289P       Coles   
10812  4583206P       Coles   
7295   2752302P       Coles   
4902   1015411P       Coles   
10297  4213819P       Coles   
634    5606330P       Coles   
2296   3209044P       Coles   
1596   4628326P       Coles   
9760   4438477P       Coles   

                                               Product Name  \
388                                Roasted & Salted Cashews   
4364                                        Natural Almonds   
8711                             Kestrel Potatoes Prepacked   
490                            Wombok Chinese Cabbage Whole   
5140                                    Garlic Bulb Prepack   
9531                                                 Celery   
2731                                 Zero Uber Thin Condoms   
9407                                    Wombok Cabbage Half   
12442       Gold+ 2 Baby Follow-On Formula From 6-12 Months   
6366     Gold+ 1 Baby Infant Formula From Birth to 6 Months   
3003                  Roasted & Salted Pistachios Prepacked   
7534                              Sugar Snap Peas Prepacked   
10812                                       Purple Eggplant   
7295                                         Salted Peanuts   
4902                            Dy Bacon Short Cut Rindless   
10297                            Brushed Potatoes Prepacked   
634                                       Topless Pineapple   
2296   Urban Coffee Culture Dolce Gusto Cappuccino Capsules   
1596                             Snow Peas Medium Prepacked   
9760                             Pink Lady Apples Prepacked   

                  Category ADL Price MEL Price Price Difference ADL packaging  \
388     Fruit & Vegetables     10.00     16.00             6.00          800g   
4364    Fruit & Vegetables     19.00     16.00             3.00          800g   
8711    Fruit & Vegetables      7.00      4.00             3.00           2kg   
490     Fruit & Vegetables      3.00      5.90             2.90        1 each   
5140    Fruit & Vegetables      0.02      2.50             2.48        4 pack   
9531   Summer Entertaining      3.50      2.00             1.50       1 bunch   
2731       Health & Beauty     15.00     16.50             1.50       20 pack   
9407    Fruit & Vegetables      2.00      3.50             1.50        1 each   
12442                 Baby     23.60     24.80             1.20          900g   
6366                  Baby     23.60     24.80             1.20          900g   
3003    Fruit & Vegetables     11.00      9.99             1.01          400g   
7534    Fruit & Vegetables      2.50      3.50             1.00          150g   
10812  Summer Entertaining      3.45      2.45             1.00        1 each   
7295    Fruit & Vegetables      5.00      4.00             1.00          400g   
4902   Dairy, Eggs & Meals      5.00      4.00             1.00          200g   
10297  Summer Entertaining      8.00      7.00             1.00           3kg   
634         Back to School      3.90      2.90             1.00        1 each   
2296                Pantry      5.50      6.50             1.00       16 pack   
1596    Fruit & Vegetables      2.50      3.50             1.00          150g   
9760        Back to School      6.50      7.50             1.00           1kg   

      MEL packaging ADL Specials MEL Specials  Same Specials  \
388            800g                                     True   
4364           800g                                     True   
8711            2kg                                     True   
490          1 each                                     True   
5140         4

In [84]:
print(f'Total list of items: {len(nonbev_diff_price_df)}')

Total list of items: 183


In [85]:
nonbev_diff_price_df.groupby('Brand')['Category'].value_counts().loc[lambda x: x>1]

Brand          Category           
Aptamil        Baby                    2
Coles          Fruit & Vegetables     30
               Summer Entertaining    16
               Back to School         15
               Pantry                  3
Colgate        Health & Beauty         2
Continental    Pantry                  6
Dettol         Health & Beauty         2
Earth Choice   Household               3
Kan Tong Thai  International Foods     2
Kleenex        Household               3
Listerine      Health & Beauty         3
Madura         Pantry                  7
MasterFoods    Pantry                  9
McKenzie's     Pantry                  2
Nerada         Pantry                  2
Nexba          Summer Entertaining     2
Nivea          Health & Beauty         2
Orgran         Pantry                  3
Swisse         Health & Beauty         2
Total Care     Pet                     3
Name: Category, dtype: int64

#### Summary Insights and Inferences

There are 712 items with different prices. After adjusting for differences in the "on special" status, there are 690 items.

The top 3 Categories account for 71.74% of the total items list:
* Liquor                 25.4%
* Drinks                 23.77%
* Summer Entertaining    22.61%

20 Brands (each of which account for more than 1% of the total) accounted for 50.58% of the list of items with different prices between Adelaide and Melbourne. The largest by far was Coles private label, at 15.65%.

Coles private label accounts for a significant proportion of products that are priced differently between Adelaide and Melbourne. This implies that Coles passes on all costs to the shoppers and their margins on these products are most likely uniform throughout Australia. This also implies that, potentially, some shoppers in certain regions of Australia are _de facto_ subsidising other shoppers when they purchase products with identical prices across different geographical locations in Australia. These products are likely not the most competitively priced in the marketplace and may be a potential area for competitor supermarkets to focus on exploiting. Another possibility is that Coles forces the suppliers to bear the "subsidies" for remote areas i.e. making a loss on the products in remote areas while making up the losses with larger profitable sales volume in metropolitan areas.

It seems most of the price difference are in the Liquor and Drinks category. The Summer Entertaining products are predominantly alcoholic and non-alcoholic drinks - Cole's API only has one label for these products, even though they can be found in the Liquor or Drinks categories on their website.

Almost all other brands are all part of the portfolio of large global food and beverage companies - companies with market caps much bigger than Coles. This may explain why they are also able to price their products differently in different regions. I wouldn't be surprised if they are the ones dictating the terms to Coles.

After filtering out products related to big beverage brands, random Colas or Sodas, and different bottles of water (Coles private label or otherwise), the biggest brand by far in the remaining list of items was Coles private brand - accounting for about 35% of that group.

#### Investigation of some oddities + commentaries

It looks like the vast majority of the products are alcohol or non-alcoholic beverages. Some of largest price differences are shocking to say the least... 

**Crazy Price Difference in a 6-pack of Coopers**

<img src="https://i.postimg.cc/YCR2xBvP/Coopers-WTF.png" />

However, I suspected that the product with the largest difference (almost 3x difference in price!!) may have been a data integrity issue. 

Afterall, it was Australia Day and someone may have knocked back a few too many too early!

I looked up Coopers Mild Ale Bottle 375ml 6-pack a few days later for both locations. My suspicions were indeed proven correct - see screenshots below (taken at the start of February):

**Adelaide**

<img src="https://i.postimg.cc/BnSrbk4m/Coopers-Mild-Ale-ADL.png" title="Adelaide Coopers Mild Ale Price"
     width="1000" height="700" />

**Melbourne**

<img src="https://i.postimg.cc/L5vVbyws/Coopers-Mild-Ale-MEL.png" title="Melbourne Coopers Mild Ale Price"
     width = "1000" height = "700" />


<br></br>

What's interesting here is the fact that the \$44 price tag is almost certainly a Liquorland price tag for a 24-pack of Coopers Mild Ale 375mL Bottle. I only had data from early January of Liquorland prices for the state of Victoria and after pulling up the data, I am almost certain that these prices are ported over from Liquorland. Does this mistake imply that some of these processes are still manual between Liquorland and Coles? Surely, they should've integrated their backend by now??

**Liquorland Coopers Mild Ale 24-pack price VIC**

<img src="https://i.postimg.cc/9MbZZnGt/liquorland-coopers-mild-ale-prices.png" />
<br></br>

As for why the vast majority of price differences are with bottle/canned beverages... I can only make some speculations. One thing that came to mind was the 10c refund label that you always find on bottles and cans. I never really looked into it but a quick look on [Wikipedia][] confirms that South Australia indeed _"charges a refundable deposit of AUD 10 cents per drinks carton, can or bottle (only containers marked as eligible for refund; does not include wine & spirits bottles, milk cartons, or concentrated/and or vegetable juice intended to be diluted before consumption)"._ It turns out this has been in place since 1977 and Victoria is expected to adopt a similar scheme by 2023, according to Wikipedia.

[Wikipedia]: https://en.wikipedia.org/wiki/Container-deposit_legislation#Australia "Wikipedia"

However, this explanation falls short in explaining all of the price differences e.g. Crown lager 24-pack would be expected to be \\$2.40 more expensive in Adelaide than in Melbourne, *ceteris paribus*, however, the actual difference is \\$4. Other products have a less than expected amount of difference e.g. Melbourne Bitters 375ml Can 6-pack only has a \\$1 difference. 

This leads me to my next questions...

I wonder if Adelaide has any Liquor/Drinks products which are cheaper than Melbourne?

If the answer is 'yes', then that would imply that Coles passes on logistics costs to the end consumers (at least in the Liquor/Drinks category).


In [106]:
"""
Define filter: 
    Price difference is not zero, 
    Price difference of the Adelaide product subtracted from the Melbourne product is positive,
    The '$pecials' tag match
"""

filter = (
    (adl_mel_df['Price Difference'] != 0) &
    (adl_mel_df['MEL Price'] - adl_mel_df['ADL Price'] > 0) &
    (adl_mel_df['Same Specials'] == True)
)

adl_mel_df.loc[filter].sort_values('Price Difference', ascending=False).head(20)


SKU             Brand  \
388    2123668P             Coles   
8235   3207774P           Coopers   
490    4879042P             Coles   
5140   3866840P             Coles   
5319   4205264P             Coles   
9407   4456832P             Coles   
3944   2580466P      Jack Daniels   
2731   9932765P        Lifestyles   
6366   6763800P           Aptamil   
12442  8900989P           Aptamil   
2296   3209044P             Coles   
9840   5950700P         Mirabella   
9760   4438477P             Coles   
7534   4619289P             Coles   
7466     10955P           Coopers   
2534     63029P           Coopers   
150    3330715P           Coopers   
12474  3207785P           Coopers   
1596   4628326P             Coles   
4475     68321P  Perfect Italiano   

                                               Product Name  \
388                                Roasted & Salted Cashews   
8235                                  Pacific Ale Can 375mL   
490                            Wombok Chinese Cabbage Whole   
5140                                    Garlic Bulb Prepack   
5319                              Watermelon Seedless Whole   
9407                                    Wombok Cabbage Half   
3944                              & No Sugar Cola Can 375mL   
2731                                 Zero Uber Thin Condoms   
6366     Gold+ 1 Baby Infant Formula From Birth to 6 Months   
12442       Gold+ 2 Baby Follow-On Formula From 6-12 Months   
2296   Urban Coffee Culture Dolce Gusto Cappuccino Capsules   
9840           Pilot Globe 15 watt Small Edison Screw Clear   
9760                             Pink Lady Apples Prepacked   
7534                              Sugar Snap Peas Prepacked   
7466                             Sparkling Ale Bottle 375mL   
2534                         Original Pale Ale Bottle 375mL   
150                                      Pale Ale Can 375mL   
12474                                 Pacific Ale Can 375mL   
1596                             Snow Peas Medium Prepacked   
4475                                                Ricotta   

                  Category ADL Price MEL Price Price Difference  \
388     Fruit & Vegetables     10.00     16.00             6.00   
8235                Liquor     58.00     63.00             5.00   
490     Fruit & Vegetables      3.00      5.90             2.90   
5140    Fruit & Vegetables      0.02      2.50             2.48   
5319    Fruit & Vegetables      9.60     12.00             2.40   
9407    Fruit & Vegetables      2.00      3.50             1.50   
3944                Liquor     23.50     25.00             1.50   
2731       Health & Beauty     15.00     16.50             1.50   
6366                  Baby     23.60     24.80             1.20   
12442                 Baby     23.60     24.80             1.20   
2296                Pantry      5.50      6.50             1.00   
9840             Household      2.25      3.25             1.00   
9760        Back to School      6.50      7.50             1.00   
7534    Fruit & Vegetables      2.50      3.50             1.00   
7466                Liquor     22.00     23.00             1.00   
2534                Liquor     20.00     21.00             1.00   
150                 Liquor     20.00     21.00             1.00   
12474               Liquor     22.00     23.00             1.00   
1596    Fruit & Vegetables      2.50      3.50             1.00   
4475   Summer Entertaining      5.30      6.25             0.95   

          ADL packaging     MEL packaging ADL Specials MEL Specials  \
388                800g              800g                             
8235            24 Pack           24 Pack                             
490              1 each            1 each                             
5140             4 pack            4 pack                             
5319   approx. 8kg each  approx. 8kg each                             
9407             1 each            1 each                             
3944 

#### Forgetting something... and revealing something interesting...

I had forgotten to ensure that both products being compared have the same 'on special' status and this revealed something pretty interesting...

Why is Coopers Original Pale Ale 24-pack and Coopers Pale Ale 24-pack in Melbourne, ON SPECIAL, more expensive than in Adelaide (NOT on special)?

<img src='https://i.postimg.cc/sxL1jRRx/speshul-indeed.png' width="600" height="400"/>

I manually grabbed screenshots after visiting the corresponding sites a few days later while conducting this analysis to see for myself. The screenshots below confirm that these two products are indeed much more expensive in Melbourne than in Adelaide. 

I looked up Coopers Brewing and it turns out they're based out of Adelaide - which can explain why a slab of Coopers is much cheaper in Adelaide than Melbourne. However, what puzzles me is the fact that a 6-pack of Coopers Mild Ale in 375mL bottles (pictured above) is cheaper in Melbourne...

**Coopers Pale Ale in Adelaide (04 FEB 2021)**
<img src='https://i.postimg.cc/tJzQk3Bj/Coopers-OG-pale-ale-ADL-normal-price.png' width="1000" height="700" />


**Coopers Pale Ale in Melbourne (04 FEB 2021)**
<img src='https://i.postimg.cc/kgG3Nt80/Coopers-OG-pale-ale-MEL-normal-price.png' width="1000" height="700" />

<br>

## \\$pecials

* What percentage of total products are on special and in what categories?
* What products are still comparatively more expensive while on special? 

From the above case, we have already established that "specials" are not uniform between states. Most likely, different products are on special at different Coles stores even within states. 



#### What percentage of products are 'on special'?

In [87]:
# Proportion of products 'on special' in both cities

print(adl_df['ADL Specials'].value_counts(normalize=True)*100)
print('\n')
print(mel_df['MEL Specials'].value_counts(normalize=True)*100)

            70.326994
$pecials    29.673006
Name: ADL Specials, dtype: float64


            69.789715
$pecials    30.210285
Name: MEL Specials, dtype: float64


#### Category breakdown

In [88]:
ald_specials = adl_df.loc[adl_df['ADL Specials'] == '$pecials']
ald_specials['Category'].value_counts(normalize=True)*100

Pantry                  23.716381
Health & Beauty         21.463500
Back to School           9.028991
Summer Entertaining      7.334963
Liquor                   6.933287
Household                6.444289
Dairy, Eggs & Meals      6.356968
Frozen                   5.029689
Pet                      2.951450
Drinks                   2.724415
International Foods      2.322738
Baby                     2.078240
Bakery                   1.414600
Meat & Seafood           1.187565
                         0.419141
Clothing                 0.314356
From The Deli            0.139714
Convenience Meals        0.104785
New                      0.017464
Chilled & Fresh Food     0.017464
Name: Category, dtype: float64

In [89]:
mel_specials = mel_df.loc[mel_df['MEL Specials'] == '$pecials']
mel_specials['Category'].value_counts(normalize=True)*100

Pantry                 28.095566
Health & Beauty        14.334941
Liquor                 11.026878
Back to School          9.441764
Summer Entertaining     8.982311
Dairy, Eggs & Meals     6.432346
Household               5.605330
Frozen                  4.181025
Drinks                  3.422927
Pet                     2.136458
International Foods     2.044567
Baby                    2.044567
Bakery                  1.056742
Meat & Seafood          0.689180
                        0.413508
From The Deli           0.045945
Clothing                0.022973
Convenience Meals       0.022973
Name: Category, dtype: float64

#### What products have mismatched '\\$pecials' tag?

In [96]:
diff_specials_df = adl_mel_df.loc[adl_mel_df['Same Specials'] == False].sort_values('MEL Specials')
diff_specials_df.sort_values('Price Difference', ascending=False)

SKU               Brand  \
12350    10078P     Victoria Bitter   
11562    10016P     Victoria Bitter   
5390   9246021P               Coles   
5025    416504P             Cascade   
4567   3330690P             Coopers   
12269    49665P             Coopers   
5705   8598420P               Coles   
7327   8600952P               Coles   
3716    311021P     Victoria Bitter   
7540    311007P     Victoria Bitter   
3792   9246065P               Coles   
8667   8970361P        Bird In Hand   
7313   8953577P         Sunny Queen   
10074  3283943P               Coles   
1056   3528712P          Alka Power   
10719  5129875P         Daily Juice   
2881   4745997P     The Daily Juice   
4330   5738160P         Daily Juice   
9676   7895431P       Kan Tong Thin   
361    8870287P    Kan Tong Hokkien   
7822   8870334P   Kan Tong Udon Wok   
2337   9322083P  Kan Tong Singapore   
10000  2244527P               Coles   
5110   1163725P               Heinz   
5423   1163689P               Heinz   
9756   3663060P         Pirate Life   
4962   3780369P      Vegie Delights   
2552   2289015P               Heinz   
2286   2860768P               Fairy   
7674   2752255P               Coles   
3259   2374191P      Great Northern   

                                                         Product Name  \
12350                                                    Bottle 375mL   
11562                                                       Can 375mL   
5390                                             Lamb Leg Roast Whole   
5025                                       Premium Light Bottle 375mL   
4567                                               Pale Ale Can 375mL   
12269                                  Original Pale Ale Bottle 375mL   
5705                                          Butcher Lamb Loin Chops   
7327                                                  Pork Spare Ribs   
3716                                                        Can 375mL   
7540                                                     Bottle 375mL   
3792                                              Lamb Leg Roast Half   
8667                                                 Pinot Rose 750mL   
7313                                    Free Range Large Eggs 18 pack   
10074                                              Brown Basmati Rice   
1056   Ionic Alkaline Spring Water PH9-10 With 5 Natural Electrolytes   
10719                                             Apple Juice Chilled   
2881                                    Breakfast Burst Juice Chilled   
4330                                   Pulp Free Orange Juice Chilled   
9676                                           Rice Wok Ready Noodles   
361                                                 Wok Ready Noodles   
7822                                                    Ready Noodles   
2337                                                Wok Ready Noodles   
10000                      Ultra Complete Clean Dishwasher 30 Tablets   
5110                      Soup of The Day Old Fashioned Chicken Pouch   
5423                          Soup Of The Day Buttercup Pumpkin & Veg   
9756                                          Pale Ale 5.4% Can 355mL   
4962                                                  Bolognese Sauce   
2552               Soup Of The Day 7 Veg With Garden Herbs Soup Pouch   
2286                     Original All In One Lemon Dishwasher Tablets   
7674                                       Nature's Kitchen Firm Tofu   
3259                                   Super Crisp Lager Bottle 330mL   

                  Category ADL Price MEL Price Price Difference  \
12350               Liquor    58.000    47.000           11.000   
11562               Liquor    58.000    48.000           10.000   
5390        Meat & Seafood    27.945    31.590            3.645   
5025                Liquor    45.000    42.000            3.000   
4567                Liquor    51.000    54.000            3.000   
12269  Summer Entertaini

#### Which products are 'on special' but are still more expensive than the identical product in the other city?

In [99]:
# Speshul pRoDucTs - on special but still more expensive than the other city's product which isn't on special

adl_diff_specials = diff_specials_df[diff_specials_df['ADL Price'] > diff_specials_df['MEL Price']].loc[diff_specials_df['ADL Specials'] == "$pecials"]
mel_diff_specials = diff_specials_df[diff_specials_df['MEL Price'] > diff_specials_df['ADL Price']].loc[diff_specials_df['MEL Specials'] == "$pecials"]

super_special_df = pd.concat([adl_diff_specials, mel_diff_specials])
super_special_df

SKU            Brand                    Product Name  \
5025    416504P          Cascade      Premium Light Bottle 375mL   
10719  5129875P      Daily Juice             Apple Juice Chilled   
12269    49665P          Coopers  Original Pale Ale Bottle 375mL   
4567   3330690P          Coopers              Pale Ale Can 375mL   
4330   5738160P      Daily Juice  Pulp Free Orange Juice Chilled   
2881   4745997P  The Daily Juice   Breakfast Burst Juice Chilled   

                  Category ADL Price MEL Price Price Difference ADL packaging  \
5025                Liquor      45.0      42.0              3.0       24 Pack   
10719               Drinks       4.8       5.0              0.2            2L   
12269  Summer Entertaining      51.0      54.0              3.0       24 Pack   
4567                Liquor      51.0      54.0              3.0       24 Pack   
4330                Drinks       4.8       5.0              0.2            2L   
2881                Drinks       4.8       5.0              0.2            2L   

      MEL packaging ADL Specials MEL Specials  Same Specials  \
5025        24 Pack     $pecials                       False   
10719            2L                  $pecials          False   
12269       24 Pack                  $pecials          False   
4567        24 Pack                  $pecials          False   
4330             2L                  $pecials          False   
2881             2L                  $pecials          False   

                                                                                                                 ADL Suppliers  \
5025                                                                                                          ['9393 CUB P/L']   
10719                                                                             ['694789 BERRI LTD', '4111 LD & D AUST P/L']   
12269  ['805291 LIQUOR CONNECTION', '79413 COOPERS BREWERY LTD', '963979 MAX LIQUOR SUPPLIES', '804668 PREMIUM BEVERAGES P/L']   
4567                                                                                            ['963979 MAX LIQUOR SUPPLIES']   
4330                                                                              ['4111 LD & D AUST P/L', '694789 BERRI LTD']   
2881                                                                                   ['694789 BERRI LTD', '4111 LD & D AUS']   

                                                                                                                 MEL Suppliers  
5025                                                                                                          ['9393 CUB P/L']  
10719                                                                             ['694789 BERRI LTD', '4111 LD & D AUST P/L']  
12269  ['805291 LIQUOR CONNECTION', '79413 COOPERS BREWERY LTD', '963979 MAX LIQUOR SUPPLIES', '804668 PREMIUM BEVERAGES P/L']  
4567                                                                                            ['963979 MAX LIQUOR SUPPLIES']  
4330                                                                              ['4111 LD & D AUST P/L', '694789 BERRI LTD']  
2881                                                                                   ['694789 BERRI LTD', '4111 LD & D AUS']